### PR TITLE
Hotfix/comments in issues

### PR DESCRIPTION
**Pull request by abdullah-siddiqui-git**
Following bugs are fixed in this PR:
-> Added comments functionality in Issues.
-> Resolved flash messages rendering issues in ajax requests.
-> Resolved theme inconsistency in app and email.
-> Resolved Signup Issue.

### CODE REVIEW

**Comment by abdullah-siddiqui-git** [comment]
Cycle - 0 review completed.

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -12,53 +14,65 @@ def new
     end
   end
 
+  # POST /projects/:project_id/comments
+  # POST /issues/:issue_id/comments
   def create
-    @comment = @project.comments.build(comment_params)
+    @comment = @commentable.comments.build(comment_params)
```
Comment (basil7vals) [inline app/controllers/comments_controller.rb]: User current_tenant

**Comment by basil7vals** [comment]
Cycle - 1 completed.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -12,57 +14,69 @@ def new
...
-      respond_to do |format|
-        format.js
-      end
-      flash[:notice] = t('comments.update.updated')
+      flash.now[:notice] = t('comments.update.updated')
     else
-      # how to show the errors
-      render 'edit'
-      flash[:notice] = t('comments.update.not_updated')
+      flash.now[:notice] = t('comments.update.not_updated')
```
Comment (Anmol79) [inline app/controllers/comments_controller.rb]: notice is for success scenarios only

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -12,57 +14,69 @@ def new
...
+    respond_to do |format|
+      format.js
     end
   end
 
   private
 
-  def load_project
-    @project = Project.find(params[:project_id])
+  def load_commentable
```
Comment (Anmol79) [inline app/controllers/comments_controller.rb]: use cancan to load through current_tenant

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -12,57 +14,69 @@ def new
...
+    if params[:project_id].present?
+      @commentable = current_tenant.projects.find(params[:project_id])
+    elsif params[:issue_id].present?
+      @commentable = current_tenant.issues.find(params[:issue_id])
+    end
   end
 
   def current_comment
-    @comment = Comment.find(params[:id])
+    @comment = current_tenant.comments.find(params[:id])
```
Comment (Anmol79) [inline app/controllers/comments_controller.rb]: use cancan

File: app/controllers/issues_controller.rb [hunk 5]
```diff
@@ -36,7 +36,10 @@ def new
 
   # GET projects/:id/issues/:id
   def show
+    @issue = current_tenant.issues.find(params[:id])
```
Comment (Anmol79) [inline app/controllers/issues_controller.rb]: cancan should load the issue

File: app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -55,7 +56,7 @@ def destroy
       flash[:notice] = t('projects.destroy.destroyed')
       redirect_to projects_path
     else
-      flash[:notice] = t('projects.destroy.not_destroyed')
+      flash.now[:notice] = t('projects.destroy.not_destroyed')
```
Comment (Anmol79) [inline app/controllers/projects_controller.rb]: incorporate this change everywhere flash.now[:error] = [t('projects.destroy.not_destroyed')]

File: app/models/company.rb [hunk 7]
```diff
@@ -2,7 +2,7 @@ class Company < ActiveRecord::Base
   not_multitenant
 
   before_save { subdomain.downcase! }
-  after_save :create_issue_states, :create_issue_types
+  after_commit :create_issue_states, :create_issue_types
```
Comment (Anmol79) [inline app/models/company.rb]: after_create

File: app/controllers/comments_controller.rb [hunk 8]
```diff
@@ -1,70 +1,72 @@
...
     respond_to do |format|
       format.js
     end
   end
 
+  # PATCH /comments/:id(.:format)
+  # PUT /comments/:id(.:format)
   def update
+    # @comment_invalid is made instance variable because it is used in update.js.erb file.
+    @comment_invalid = false
```
Comment (abubakarazeem) [inline app/controllers/comments_controller.rb]: Variable name should be changed.

File: app/controllers/comments_controller.rb [hunk 9]
```diff
@@ -1,70 +1,72 @@
...
     if @comment.update(comment_params)
-      respond_to do |format|
-        format.js
-      end
-      flash[:notice] = t('comments.update.updated')
+      flash.now[:notice] = t('comments.update.updated')
```
Comment (abubakarazeem) [inline app/controllers/comments_controller.rb]: Improve reference to yml messages.

File: app/controllers/projects_controller.rb [hunk 10]
```diff
@@ -16,7 +16,8 @@ def new
 
   def create
     if @project.save
-      redirect_to @project, notice: t('projects.create.created')
+      flash[:notice] = t('projects.create.created')
+      redirect_to @project
```
Comment (abubakarazeem) [inline app/controllers/projects_controller.rb]: study return if_performed?

File: app/controllers/projects_controller.rb [hunk 11]
```diff
@@ -58,7 +59,8 @@ def destroy
       if @project.destroyed?
```
Comment (abubakarazeem) [inline app/controllers/projects_controller.rb]: Study difference between .destroy and .destroyed?

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 12]
```diff
@@ -162,7 +162,7 @@ a[x-apple-data-detectors] {
                   <td class="es-m-p0r es-m-p20b" width="178" valign="top" align="center" style="padding:0;Margin:0;"> 
                    <table width="100%" cellspacing="0" cellpadding="0" style="mso-table-lspace:0pt;mso-table-rspace:0pt;border-collapse:collapse;border-spacing:0px;"> 
                      <tr style="border-collapse:collapse;"> 
-                      <td align="center" style="padding:0;Margin:0;"> <img class="adapt-img" src="https://gdypl.stripocdn.email/content/guids/dd93cbd4-9fa9-4533-badb-c3e2baf9857f/images/62371564991504631.jpg" alt style="display:block;border:0;outline:none;text-decoration:none;-ms-interpolation-mode:bicubic;" width="178"></td> 
+                      <td align="center" style="padding:0;Margin:0;"> <img class="adapt-img" src="https://gdypl.stripocdn.email/content/guids/dd93cbd4-9fa9-4533-badb-c3e2baf9857f/images/62371564991504631.jpg" alt style="display:block;border:0;outline:none;text-decoration:none;-ms-interpolation-mode:bicubic;border-radius: 20%;" width="178"></td>
```
Comment (abubakarazeem) [inline app/views/devise/mailer/confirmation_instructions.html.erb]: Why use style attributes in email template?

File: app/views/issue_watchers/search_watcher_to_destroy.js.erb [hunk 13]
```diff
@@ -1,3 +1,3 @@
 $('#remove_modal_search_results').html('')
-  .html("<%= j render partial: 'watchers_search_result', locals: { watchers: @watchers, issue: @issue, action: 'destroy' } %>")  
-  
\ No newline at end of file
+  .html("<%= j render partial: 'watchers_search_result', locals: { watchers: @watchers, issue: @issue, action: 'destroy' } %>");  
+$("#flash_messages").html("<%= j render partial: 'shared/flash' %>");
```
Comment (abubakarazeem) [inline app/views/issue_watchers/search_watcher_to_destroy.js.erb]: Study difference between sanitize, to_json, html_safe, raw, h

File: app/controllers/projects_controller.rb (hunk 6 above)
Comment (MidhatAsif) [reply app/controllers/projects_controller.rb]: done

File: app/controllers/comments_controller.rb (hunk 9 above)
Comment (MidhatAsif) [reply app/controllers/comments_controller.rb]: done

File: app/controllers/comments_controller.rb (hunk 8 above)
Comment (MidhatAsif) [reply app/controllers/comments_controller.rb]: done

File: app/controllers/projects_controller.rb (hunk 10 above)
Comment (MidhatAsif) [reply app/controllers/projects_controller.rb]: done

File: app/controllers/projects_controller.rb (hunk 11 above)
Comment (MidhatAsif) [reply app/controllers/projects_controller.rb]: done

**Pull request merged**